### PR TITLE
Check rsync version

### PR DESCRIPTION
**Pull request by kevinboulain**
Hi there,

Gave a quick shot at checking rsync's version for #5.

It's probably not the best solution (a better one would be to fallback on `--progress` depending on the version for example) but I have some questions now:
 - why not always use `--progress` (is it because `--info=progress2` is prettier :))?
 - `--info=progress2` has been introduced more than 5 years ago, isn't a fallback good enough?
 - would you introduce a new `tool` (something like `oldRsync` for example) for an easier fallback to `--progress` (seems a bit overkill)?

Disclaimer: I lack Go experience, feel free to criticize.

### CODE REVIEW

File: cmd/push/main.go [hunk 1]
```diff
@@ -95,10 +97,34 @@ func (t tool) push(verbose bool, src string, pkgs []string, host, rel string) er
...
+	}
+	major, err := strconv.Atoi(string(matches[1]))
+	if err != nil {
+		return false
+	}
+	minor, err := strconv.Atoi(string(matches[2]))
+	if err != nil {
+		return false
+	}
+	// --info=progress2 has been introduced in 3.1.0
```
Comment (maruel) [inline cmd/push/main.go:138]: I'd prefer instead:

- `getRsyncVersion() (int, int)` that returns major and minor version, and 0, 0 if none.
- in detect, return `rsyncOld` or `rsyncRecent` based on the minimum version
- On rsyncOld, skip the flag --info-progress2.

**Comment by kevinboulain** [comment]
Sorry for the delay, was away.

I didn't do exactly as you said (to help avoid redundancy in detection), but it's pretty similar.
I looked into `--out-format` as an alternative to variants but it didn't seem really convincing, feel free to discard if you find a format that suites you for all versions.

Should fix #5.

**Comment by maruel** [comment]
Oh I thought you had abandoned it so I made a change a few days ago but hadn't pushed to master yet as I wanted to test locally on my mac first;
https://github.com/periph/bootstrap/compare/rsync

I prefer the approach I used with regards to managing arguments, and I updated detection to be nicer.

**Comment by maruel** [comment]
Subsumed by https://github.com/periph/bootstrap/commit/daabf82d3a0a650ef939c0c1852925afdb5eba4a

Still, thanks for the change, it forced me to take the time to look at the problem and fix it.

**Comment by kevinboulain** [comment]
No worries.

Can I ask what's with the `toolIndex` (why not use a switch or map), it seems a bit weird to me?

**Comment by maruel** [comment]
This form is generated by https://godoc.org/golang.org/x/tools/cmd/stringer, which optimizes for both performance and memory usage.

**Comment by maruel** [comment]
but I formatted it manually.

**Pull request closed without merging**